drivers/hmis/app/graphql/types/hmis_schema/query_type.rb

```diff
@@ -235,7 +241,12 @@ def service_form_definition(service_type_id:, project_id:)
       service_type = Hmis::Hud::CustomServiceType.find_by(id: service_type_id)
       raise HmisErrors::ApiError, 'Service type not found' unless service_type.present?
 
-      Hmis::Form::Definition.find_definition_for_service_type(service_type, project: project)
+      definition = Hmis::Form::Definition.find_definition_for_service_type(service_type, project: project)
+
+      # Similar to record_form_definition above, we always want to return a definition if we possibly can, so use the
+      # HUD default form. For non-HUD services, the only field that would show up is Date Provided.
+      # TODO(#6763) Use "system_managed" flag when it exists
+      definition || Hmis::Form::Definition.with_role(:SERVICE).where(identifier: 'service').first
```

Comment:
I have thought this through a little more @martha and maybe come to a different conclusion. Given that we don't have a "default" form that works for all Services, I think we should just revert this to the old behavior where it returns empty (and subsequently errors) if no form is found for a CustomService type. 

```suggestion
      # Similar to record_form_definition above, we always want to return a definition if we possibly can, so use the
      # default HUD Service form for HUD service types. For Custom service types, return empty because we can't determine which form to use.
      # TODO(#6763) Use "system_managed" flag when it exists
      definition || Hmis::Form::Definition.with_role(:SERVICE).where(identifier: 'service').first if service_type.hud_service?
```



[This is the frontend code](https://github.com/greenriver/hmis-frontend/blob/ccf0f5de4a293ca3f301eaba78290e981a8fd92e/src/modules/services/hooks/useServiceDialog.tsx#L114-L118) that hits this query. There are 2 cases:
1) User is creating a new service, and they selected a type in the dropdown (`AVAILABLE_SERVICE_TYPES`). In this case we expect the form to exist, and its OK to error if it doesn't.
2) User is editing an existing Custom Service. It should have a form processor definition_id, because CustomServices are generated in HMIS, or they are migrated-in during which we'd add a FormProcessor that points to the correct form.
3) User is editing an existing HUD Service. Default to the HUD form as suggested.


(1) should be functional already, but I added https://github.com/open-path/Green-River/issues/6939 as an idea to simplify it.
(2) won't really be real until https://github.com/open-path/Green-River/issues/6113 is implemented, but I think it's OK to retain current behavior in the mean time (where it expects the service to still be enabled).
(3) is achieved by the diff

What do you think?